modules/configure_project/main.tf

```diff
@@ -0,0 +1,102 @@
+  read_path      = "${local.dataplatform_api}{id}"
+  read_method    = "GET"
+  create_path    = "${local.dataplatform_api}/transactional/v2/projects?verify_unique_name=true"
+  create_method  = "POST"
+  id_attribute   = "location"
+  destroy_method = "DELETE"
+  destroy_path   = "${local.dataplatform_api}/transactional{id}"
+  data           = <<-EOT
+                  {
+                    "name": "${var.watsonx_project_name}",
+                    "generator": "watsonx-saas-da",
+                    "type": "wx",
+                    "storage": {
+                      "type": "bmcos_object_storage",
+                      "guid": "${var.cos_guid}",
+                      "resource_crn": "${var.cos_crn}",
+                      "delegated": ${var.watsonx_project_delegated}
+                    },
+                    "description": "${var.watsonx_project_description}",
+                    "public": true,
+                    "tags": ${jsonencode(var.watsonx_project_tags)},
+                    "compute": [
+                      {
+                        "name": "${var.machine_learning_name}",
+                        "guid": "${var.machine_learning_guid}",
+                        "type": "machine_learning",
+                        "crn": "${var.machine_learning_crn}"
+                      }
+                    ],
+                    "settings": {
+                      "access_restrictions": {
+                        "data": ${var.watsonx_mark_as_sensitive}
+                      }
+                    }
+                  }
+                  EOT
+  update_method  = "PATCH"
+  update_path    = "${local.dataplatform_api}{id}"
+  update_data    = <<-EOT
+                  {
+                    "name": "${var.watsonx_project_name}",
+                    "type": "wx",
+                    "description": "${var.watsonx_project_description}",
+                    "public": true,
+                    "compute": [
+                      {
+                        "name": "${var.machine_learning_name}",
+                        "guid": "${var.machine_learning_guid}",
+                        "type": "machine_learning",
+                        "crn": "${var.machine_learning_crn}",
+                        "credentials": {}
+                      }
+                    ]
+                  }
+                  EOT
+}
+
+resource "time_sleep" "wait_5_seconds" {
+  depends_on      = [restapi_object.configure_project]
+  create_duration = "5s"
+}
+
+data "restapi_object" "get_project" {
+  depends_on   = [resource.restapi_object.configure_project, resource.time_sleep.wait_5_seconds]
+  path         = "${local.dataplatform_api}/v2/projects"
+  query_string = "project_ids=${local.watsonx_project_id}"
+  results_key  = "resources"
+  search_key   = "metadata/guid"
+  search_value = local.watsonx_project_id
+  id_attribute = "metadata/guid"
+}
+
+locals {
+  dataplatform_api_mapping = {
+    "us-south" = "//api.dataplatform.cloud.ibm.com",
```

Comment:
Missing `https` here. Referred [this] page(https://cloud.ibm.com/apidocs/watsonx-ai#endpoint-url)

Reply:
This one was copied from the existing DA code. So I am not sure if `https` is required here...it is used to create restapi object. UI is accessible on `https` endpoint only.